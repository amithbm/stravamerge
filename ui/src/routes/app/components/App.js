import React from 'react';
import HeaderContainer from '../containers/HeaderContainer'
import InProgressBarContainer from '../../common/containers/InProgressBarContainer'

export default class App extends React.Component {
  constructor(props, context) {
    super(props)
    this.router = context.router
  }

  componentDidMount() {
    if (this.props.isAuthenticated && this.props.location.pathname === '/') {
      this.router.push('/dashboard')
    }
  }

  render() {
    return (
      <div>
        <HeaderContainer></HeaderContainer>
        <InProgressBarContainer></InProgressBarContainer>
          { this.props.children }
      </div>
    );
  }
}

App.contextTypes = {
    router: React.PropTypes.object.isRequired
};

