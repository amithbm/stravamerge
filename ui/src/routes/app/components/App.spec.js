import React from 'react'
import App from './App'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('components', () => {
    describe('<App />', () => {
        it('renders as expected', () => {
            // the below illustrates how to deal with components that require items in the React context
            const context = { router: { }}
            const component = shallow(<App isAuthenticated={true} />, { context })
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})
