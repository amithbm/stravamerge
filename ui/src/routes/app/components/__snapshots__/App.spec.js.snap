exports[`components <App /> renders as expected 1`] = `
<div>
  <Connect(Component) />
  <Connect(InProgressBar) />
</div>
`;
